src/pages/set/_.vue

```diff
@@ -174,6 +176,15 @@
       SetRecommendations: () => import('@/components/set/SetRecommendations')
     },
 
+    async beforeRouteEnter(to, from, next) {
```

Comment:
This should be a middleware instead so that it happens earlier and can stop the rendering of the page. As it is here, there is a brief 404 error page visible, then it redirects away.